src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServer.cs

```diff
@@ -119,14 +119,36 @@ public class KestrelServer : IServer
                 var hasListenOptions = listenOptions.Any();
                 var hasServerAddresses = _serverAddresses.Addresses.Any();
 
-                if (hasListenOptions && hasServerAddresses)
+                if (_serverAddresses.PreferHostingUrls && hasServerAddresses)
                 {
-                    var joined = string.Join(", ", _serverAddresses.Addresses);
-                    _logger.LogWarning($"Overriding address(es) '{joined}'. Binding to endpoints defined in UseKestrel() instead.");
+                    if (hasListenOptions)
+                    {
+                        var joined = string.Join(", ", _serverAddresses.Addresses);
+                        _logger.LogWarning($"Overriding endpoints defined in UseKestrel() since {nameof(IServerAddressesFeature.PreferHostingUrls)} is set to true. Binding to address(es) '{joined}' instead.");
```

Comment:
This is something that we expect to happen when PreferHostingUrls is set to true. Is this worth a warning? Or should we log as info?